pkg/agent/controller/networkpolicy/reconciler.go

```diff
@@ -504,17 +503,25 @@ func podsToOFAddresses(podSet v1beta1.GroupMemberPodSet) []types.Address {
 	return addresses
 }
 
-func ipBlocksToOFAddresses(ipBlocks []v1beta1.IPBlock) ([]types.Address, []types.Address) {
+func ipBlocksToOFAddresses(ipBlocks []v1beta1.IPBlock) []types.Address {
 	// Must not return nil as it means not restricted by addresses in Openflow implementation.
-	addresses := make([]types.Address, 0, len(ipBlocks))
-	var exceptAddresses []types.Address
+	addresses := make([]types.Address, 0)
 	for _, b := range ipBlocks {
-		addresses = append(addresses, ipNetToOFAddress(b.CIDR))
-		for _, e := range b.Except {
-			exceptAddresses = append(exceptAddresses, ipNetToOFAddress(e))
+		exceptIPNet := make([]*net.IPNet, 0, len(b.Except))
+		for _, c := range b.Except {
+			exceptIPNet = append(exceptIPNet, ip.IPNetToNetIPNet(&c))
+		}
+		diffCIDRs, err := ip.DiffFromCIDRs(ip.IPNetToNetIPNet(&b.CIDR), exceptIPNet)
+		if err != nil {
+			// Currently only IPv4 addresses are supported
+			klog.Errorf("Error when determining diffCIDRs: %v", err)
```

Comment:
should you add a `continue` statement here, since `diffCIDRs` may be invalid in case of error